src/Models/ElementalArea.php

```diff
@@ -98,105 +118,103 @@ public function renderSearchContent()
+            return DBField::create_field('HTMLText', sprintf(
+                '<a href="%s">%s</a>',
+                $owner->CMSEditLink(), $owner->Title
+            ));
         }
     }
 
     /**
-     * @return HasManyList
-     */
-    public function ItemsToRender()
-    {
-        return $this->Elements(array(
-            'Enabled' => 1
-        ));
-    }
-
-    /**
-     * Used in template instead of {@link Elements()} to wrap each element in its
-     * controller, making it easier to access and process form logic and
+     * Used in template instead of {@link Elements()} to wrap each element in
+     * its' controller, making it easier to access and process form logic and
      * actions stored in {@link ElementController}.
      *
      * @return ArrayList - Collection of {@link ElementController} instances.
      */
     public function ElementControllers()
     {
         $controllers = new ArrayList();
-        $items = $this->ItemsToRender();
+        $items = $this->Elements();
+
         if (!is_null($items)) {
             foreach ($items as $element) {
                 $controller = $element->getController();
                 $controllers->push($controller);
             }
         }
+
         return $controllers;
     }
 
     /**
-    * Return an ArrayList of pages with the Element Page Extension
-    *
-    * @return ArrayList
-    */
+     *
+     */
     public function getOwnerPage()
     {
         if ($this->OwnerClassName) {
             $class = $this->OwnerClassName;
-            $elementalAreaRelations = ElementalAreasExtension::get_elemental_area_relations(singleton($class));
+            $elementalAreaRelations = Injector::inst()->get($class)->getElementalRelations();
+
             foreach ($elementalAreaRelations as $eaRelationship) {
                 $areaID = $eaRelationship . 'ID';
+
                 $page = $class::get()->filter($areaID, $this->ID);
+
                 if ($page && $page->exists()) {
                     return $page->first();
                 }
             }
         }
 
         $originalMode = Versioned::get_stage();
+
         if (!$originalMode) {
             $originalMode = Versioned::DRAFT;
         }
+
         Versioned::set_stage(Versioned::DRAFT);
-        $elementalPageTypes = self::elemental_page_types();
-        foreach ($elementalPageTypes as $elementalPageType) {
-            $elementalAreaRelations = ElementalAreasExtension::get_elemental_area_relations(singleton($elementalPageType));
+
+        foreach ($this->supportedPageTypes() as $class) {
+            $elementalAreaRelations = Injector::inst()->get($class)->getElementalRelations();
+
             foreach ($elementalAreaRelations as $eaRelationship) {
                 $areaID = $eaRelationship . 'ID';
-                $page = $elementalPageType::get()->filter($areaID, $this->ID);
+                $page = $class::get()->filter($areaID, $this->ID);
```

Comment:
`$class` could be undefined at this point - can we tidy the logic up a bit so it's clearer that this wouldn't happen?

Reply:
Hmm, $class is in a if so pretty sure it can't be undefined but agree, this is a bit meh.

Reply:
I only skim read, but thought the condition where it's assigned is not the same as this section